native/src/routes/SearchModal.tsx

```diff
@@ -37,51 +39,76 @@ const Wrapper = styled.View`
 `
 
 export type SearchModalProps = {
-  categories: CategoriesMapModel
+  allPossibleResults: Array<SearchResult>
   navigateTo: (routeInformation: RouteInformationType) => void
   theme: ThemeType
   languageCode: string
   cityCode: string
   closeModal: (query: string) => void
   t: TFunction<'search'>
   initialSearchText: string
+  loading: boolean
 }
 
 const SearchModal = ({
-  categories,
+  allPossibleResults,
   navigateTo,
   theme,
   languageCode,
   cityCode,
   closeModal,
   t,
   initialSearchText = '',
+  loading,
 }: SearchModalProps): ReactElement => {
   const [query, setQuery] = useState<string>(initialSearchText)
   const resourceCache = useResourceCache({ cityCode, languageCode })
 
-  const searchResults = useMemo(() => searchCategories(categories, query), [categories, query])
-
-  const onItemPress = (category: CategoryModel): void => {
-    const routeInformation: CategoriesRouteInformationType = {
-      route: CATEGORIES_ROUTE,
-      cityContentPath: category.path,
-      cityCode,
-      languageCode,
-    }
-    sendTrackingSignal({
-      signal: {
-        name: SEARCH_FINISHED_SIGNAL_NAME,
-        query,
-        url: urlFromRouteInformation(routeInformation),
+  const minisearch = useMemo(() => {
+    const search = new MiniSearch({
+      fields: ['title', 'content'],
+      storeFields: ['title', 'content', 'path', 'location', 'url', 'thumbnail'],
+      searchOptions: {
+        boost: { title: 2 },
+        fuzzy: true,
+        prefix: true,
       },
     })
-    navigateTo({
-      route: CATEGORIES_ROUTE,
-      cityCode,
-      languageCode,
-      cityContentPath: category.path,
-    })
+    search.addAll(allPossibleResults)
+    return search
+  }, [allPossibleResults])
+
+  // Minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
+  const searchResults = minisearch.search(query) as unknown as SearchResult[]
+
+  const showSnackbar = useSnackbar()
+
+  const followLink = (link: string, isExternalUrl: boolean): void => {
+    if (isExternalUrl) {
+      sendTrackingSignal({
+        signal: {
+          name: SEARCH_FINISHED_SIGNAL_NAME,
+          query,
+          url: link,
+        },
+      })
+      openExternalUrl(link, showSnackbar)
+    } else {
+      const routeInformation = new InternalPathnameParser(link, languageCode, cityCode).route()
+      if (!routeInformation) {
+        reportError('no routeInformation found')
+        showSnackbar({ text: 'unknownError' })
+        return
+      }
+      sendTrackingSignal({
+        signal: {
+          name: SEARCH_FINISHED_SIGNAL_NAME,
+          query,
+          url: urlFromRouteInformation(routeInformation),
+        },
+      })
+      navigateTo(routeInformation)
+    }
   }
```

Comment:
You can just use `useNavigateToLink` instead which should do exactly what you need